integrationtest/mapper/pojo/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/IndexedEmbeddedBaseIT.java

```diff
@@ -657,18 +657,12 @@ public String getInvalid() {
 			}
 		}
 
-		backendMock.expectSchema( INDEX_NAME, b -> b
-				.field( "text", String.class )
-				// TODO HSEARCH-899 this isn't supposed to be legal => throw an exception for
-				.objectField( "invalid", c -> {
-				} )
-		);
-
-		JavaBeanMapping mapping = setupHelper.start()
+		SubTest.expectException( () -> setupHelper.start()
```

Comment:
I think it would be clearer if you set these expectations right from the start, in the commit that adds the test. But to each his own, I guess.

Reply:
Squashed with the subsequent one.